phpBB/viewforum.php

```diff
@@ -285,6 +285,20 @@
 $sort_by_text = array('a' => $user->lang['AUTHOR'], 't' => $user->lang['POST_TIME'], 'r' => $user->lang['REPLIES'], 's' => $user->lang['SUBJECT'], 'v' => $user->lang['VIEWS']);
 $sort_by_sql = array('a' => 't.topic_first_poster_name', 't' => array('t.topic_last_post_time', 't.topic_last_post_id'), 'r' => (($auth->acl_get('m_approve', $forum_id)) ? 't.topic_posts_approved + t.topic_posts_unapproved + t.topic_posts_softdeleted' : 't.topic_posts_approved'), 's' => 'LOWER(t.topic_title)', 'v' => 't.topic_views');
 
+/**
+ * Modify the topic ordering if needed
+ *
+ * @event core.viewforum_modify_topic_ordering
+ * @var array	sort_by_text	Topic ordering options
+ * @var array	sort_by_sql		Topic ordering options
```

Comment:
`Topic orderings options SQL equivalent`?

Reply:
`			$event['sort_by_sql'] = array(
				'a'	=> array('t.topic_priority', 't.topic_first_poster_name'),
				't'	=> array('t.topic_priority', 't.topic_last_post_time', 't.topic_last_post_id'),
				'r'	=> ( ($this->auth->acl_get('m_approve', $forum_id)) ? array('t.topic_priority', 't.topic_posts_approved + t.topic_posts_unapproved + t.topic_posts_softdeleted') : array('t.topic_priority', 't.topic_posts_approved') ),
				's'	=> array('t.topic_priority', 'LOWER(t.topic_title)'),
				'v'	=> array('t.topic_priority', 't.topic_views'),
			);`

Reply:
@marc1706 I am overall prepending the newly created `topic_priority` to "cement" topics of any sort, due to an extension of mine.

Reply:
I understand that. My point was that it sounds to me like the description of sort_by_text and sort_by_sql should be different. While sort_by_text is a map between the options and their respective options, sort_by_sql is a map between the options and their SQL ordering equivalent. Hence why I suggested this change.

Reply:
Oh, I didn't got your point. I assumed you wanted to see the usage of it instead.

Pls, at this point feel free to amend the description if you so wish.

Reply:
I will do it, I see has been not yet rebased. ;))